grai-integrations/source-bigquery/src/grai_source_bigquery/base.py

```diff
@@ -26,16 +30,23 @@ def get_nodes_and_edges(connector: BigqueryConnector, version: Literal["v1"]) ->
     with connector.connect() as conn:
         nodes, edges = conn.get_nodes_and_edges()
 
+    log_edges = []
+
+    if logging_connector is not None:
+        with logging_connector.connect() as conn:
+            log_edges = conn.get_edges(nodes)
+
     nodes = adapt_to_client(nodes, version)
-    edges = adapt_to_client(edges, version)
+    edges = adapt_to_client(edges + log_edges, version)
```

Comment:
I'm a little worried about structuring it this way without some form of authentication logic. If the window is wide (or the timing bad) `log_edges` could produce references to non-existent nodes which would error during the update.

I would consider moving this logic inside of a `get_edges` method on the `LoggingConnector` again to keep the interfaces clean. 

If the LoggingConnector has access to the BigqueryConnector cache it can also efficiently verify the source/destination nodes exist in the node list. There are perhaps two tests we want to run here:

1. all(edge.source in {nodes}) and all(edge.destination in {nodes})
2. len(edge) == len(set(edge.unique_id))


(2) would violate one of our uniqueness constraints in the database.

Reply:
This logic already exists within the `LoggingConnector` however subclassing `BigQueryConnector` might make sense, so that you have exactly the same credentials etc.